code/obj/machinery/computer/QM_supply.dm

```diff
@@ -530,11 +530,14 @@ var/global/datum/cdc_contact_controller/QM_CDC = new()
 					var/datum/supply_packs/P = O.object
 					shippingmarket.supply_requests -= O
 					if(wagesystem.shipping_budget >= P.cost)
-						wagesystem.shipping_budget -= P.cost
 						O.object = P
 						O.orderedby = usr.name
 						var/default_comment = ""
-						O.comment = copytext(html_encode(tgui_input_text(usr, "Comment:", "Enter comment", default_comment, multiline = TRUE)), 1, MAX_MESSAGE_LEN)
+						O.comment = tgui_input_text(usr, "Comment:", "Enter comment", default_comment, multiline = TRUE, max_length = 32, allowEmpty = TRUE)
+						if (O.comment == null)
+							return .("list")// The user cancelled the order
```

Comment:
this is so jank

Reply:
I'll try to ease the jank with the power of macros.

Reply:
There we go. Tell me if it's worse or better.

Reply:
oh i mean `.("list")` lol

Reply:
but yes those changes r good